### PR TITLE
fix: replay fix dependency resolution

### PR DESCRIPTION
**Pull request by SumitAgrawal03071989**
fixes: #632

### CODE REVIEW

**Review by arinda-arif — COMMENTED** [review]
we can go with this change to fix the issue, however, in every replay & backup request, there will be calls to external Optimus to check for each of the unresolved dependencies. we can monitor the performance and later improve this.
fyi @sravankorumilli

File: job/service.go [hunk 1]
```diff
@@ -481,13 +481,21 @@ func (srv *Service) GetDownstream(ctx context.Context, projectSpec models.Projec
 	return jobSpecs, nil
 }
 
-func (srv *Service) prepareJobSpecMap(ctx context.Context, projectSpec models.ProjectSpec) (map[string]models.JobSpec, error) {
-	// resolve dependency of all jobs in given project
-	jobSpecs, err := srv.GetDependencyResolvedSpecs(ctx, projectSpec, nil)
+func (srv *Service) prepareJobSpecMap(ctx context.Context, projectSpec models.ProjectSpec) (jobSpec map[string]models.JobSpec, resolvedErrors error) {
+	//// resolve dependency of all jobs in given project
```
[the remark this answers is not in the document]
Comment (SumitAgrawal03071989) [reply job/service.go]: Updated PR with this change

File: job/service.go [hunk 2]
```diff
@@ -481,13 +481,21 @@ func (srv *Service) GetDownstream(ctx context.Context, projectSpec models.Projec
 	return jobSpecs, nil
 }
 
-func (srv *Service) prepareJobSpecMap(ctx context.Context, projectSpec models.ProjectSpec) (map[string]models.JobSpec, error) {
-	// resolve dependency of all jobs in given project
-	jobSpecs, err := srv.GetDependencyResolvedSpecs(ctx, projectSpec, nil)
```
[the remark this answers is not in the document]
Comment (SumitAgrawal03071989) [reply job/service.go:486]: Updated PR with this change

**Comment by sravankorumilli** [comment]
> 

@arinda-arif or @SumitAgrawal03071989 after this does it replay for the external jobs ??

**Comment by sravankorumilli** [comment]
@SumitAgrawal03071989 can you link the issue in the pr

**Pull request merged**